src/cargo/ops/cargo_test.rs

```diff
@@ -137,48 +138,44 @@ fn run_doc_tests(
 ) -> CargoResult<(Test, Vec<ProcessError>)> {
     let mut errors = Vec::new();
 
-    // The unstable doctest-xcompile feature enables both per-target-ignores and
-    // cross-compiling doctests. As a side effect, this feature also gates running
-    // doctests with runtools when target == host.
-    let doctest_xcompile = config.cli_unstable().doctest_xcompile;
-    let mut runtool: &Option<(std::path::PathBuf, Vec<String>)> = &None;
-    if doctest_xcompile {
-        runtool = compilation.target_runner();
-    } else if compilation.host != compilation.target {
-        return Ok((Test::Doc, errors));
-    }
-
     for doctest_info in &compilation.to_doc_test {
         let Doctest {
-            package,
-            target,
             args,
             unstable_opts,
+            unit,
         } = doctest_info;
-        config.shell().status("Doc-tests", target.name())?;
-        let mut p = compilation.rustdoc_process(package, target)?;
+
+        // Skip any `--target` tests unless `doctest-xcompile` is specified.
+        if !config.cli_unstable().doctest_xcompile && !unit.kind.is_host() {
```

Comment:
I think the failing test is because the old code used to do `compilation.host != compilation.target` instead of `is_host()` here.

Reply:
Er oops sorry I read that nightly tests were failing and then completely forgot to actually address the issue...